action/action-dsl/src/main/java/com/powsybl/action/dsl/GroovyDslContingenciesProviderFactory.java

```diff
@@ -25,10 +26,22 @@ public GroovyDslContingenciesProvider create() {
         return new GroovyDslContingenciesProvider(dslFile);
     }
 
+    /**
+     * Creates a provider which will read the DSL from a UTF-8 encoded file.
+     */
     @Override
     public GroovyDslContingenciesProvider create(Path dslFile) {
         Objects.requireNonNull(dslFile);
         return new GroovyDslContingenciesProvider(dslFile);
     }
 
+    /**
+     * Creates a provider which will read the DSL from a UTF-8 encoded input stream.
+     */
+    @Override
+    public GroovyDslContingenciesProvider create(InputStream dslStream) {
+        Objects.requireNonNull(dslStream);
```

Comment:
Remove this check: already done by the GroovyDslContingencies constructor. Please, can you also adapt the create(Path) method